### PR TITLE
General announcements for school, route, all + success/error messages 

### PR DESCRIPTION
**Pull request by ferncorona**
Still need to do route announcements

### CODE REVIEW

File: backend/ht_buses_app/views/announcements/announcements.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from rest_framework.decorators import api_view, permission_classes
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework.permissions import IsAdminUser, AllowAny
+from rest_framework.response import Response
+from . import announcement_tools
+from django.conf import settings
+
+@csrf_exempt
+@api_view(["POST"])
+@permission_classes([AllowAny])
```
Comment (eve658) [inline backend/ht_buses_app/views/announcements/announcements.py:12]: This should be switched to `IsAdminUser`

File: backend/ht_buses_app/views/announcements/announcements.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return Response(data)
+    except:
+        data["message"] = "no users found"
+        data["success"] = False
+        return Response(data, status = 404) 
+
+
+@csrf_exempt
+@api_view(["POST"])
+@permission_classes([AllowAny])
```
Comment (eve658) [inline backend/ht_buses_app/views/announcements/announcements.py:28]: This should be switched to `IsAdminUser`

File: backend/ht_buses_app/views/announcements/announcements.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        data = announcement_tools.send_mass_announcement(subject, body, recipients, include_route_info)
+        return Response(data)
+    except:
+        data["message"] = "invalid school id"
+        data["success"] = False
+        return Response(data, status = 404) 
+
+@csrf_exempt
+@api_view(["POST"])
+@permission_classes([AllowAny])
```
Comment (eve658) [inline backend/ht_buses_app/views/announcements/announcements.py:46]: This should be switched to `IsAdminUser`

**Review by eve658 — APPROVED** [review]
lgtm :)

**Review by kyratichan — APPROVED** [review]
lgtm

**Pull request merged**